File: app/schemas/task.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional
from datetime import datetime
from enum import Enum

class Priority(str, Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    URGENT = "URGENT"

class TaskBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=500)
    description: Optional[str] = Field(None, max_length=2000)
    priority: Priority = Priority.MEDIUM
    deadline: Optional[datetime] = None

class TaskCreate(TaskBase):
    pass

class TaskUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=500)
    description: Optional[str] = Field(None, max_length=2000)
    priority: Optional[Priority] = None
    deadline: Optional[datetime] = None
    is_completed: Optional[bool] = None

class TaskResponse(TaskBase):
    id: int
    user_id: str
    is_completed: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class TaskListResponse(BaseModel):
    success: bool = True
    data: list[TaskResponse]
    total: int
    completed: int
    pending: int

class TaskSingleResponse(BaseModel):
    success: bool = True
    data: TaskResponse

class TaskStatsResponse(BaseModel):
    success: bool = True
    data: dict = {
        "total": int,
        "completed": int,
        "pending": int,
        "overdue": int,
        "by_priority": dict
    }

File: android/app/src/main/java/com/moodmate/data/model/Task.kt
package com.moodmate.data.model

data class TaskCreate(
    val title: String,
    val description: String? = null,
    val priority: String = "MEDIUM", // LOW, MEDIUM, HIGH, URGENT
    val deadline: String? = null // ISO format
)

data class TaskUpdate(
    val title: String? = null,
    val description: String? = null,
    val priority: String? = null,
    val deadline: String? = null,
    val is_completed: Boolean? = null
)

data class TaskResponse(
    val id: Int,
    val user_id: String,
    val title: String,
    val description: String? = null,
    val priority: String,
    val deadline: String? = null,
    val is_completed: Boolean,
    val created_at: String,
    val updated_at: String
)

data class TaskListResponse(
    val success: Boolean,
    val data: List<TaskResponse>,
    val total: Int,
    val completed: Int,
    val pending: Int
)

data class TaskSingleResponse(
    val success: Boolean,
    val data: TaskResponse
)

data class TaskStatsResponse(
    val success: Boolean,
    val data: TaskStats
)

data class TaskStats(
    val total: Int,
    val completed: Int,
    val pending: Int,
    val overdue: Int,
    val by_priority: Map<String, Int>
)

